tools/clang_format.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import argparse
+import logging
+import os
+import re
+import subprocess
+import sys
+import tempfile
+import typing
+
+import githelper
+
+
+# We recommend a maximum line length of 80, but do allow up to 100 characters
+# if deemed necessary by the developer. Lines that exceed that limit will
+# be wrapped after 80 characters automatically.
+LINE_LENGTH_THRESHOLD = 100
+BREAK_BEFORE = 80
+
+
+def get_clang_format_config_with_columnlimit(rootdir, limit):
+    cpp_file = os.path.join(rootdir, "src/mixxx.cpp")
+    proc = subprocess.run(
+        ["clang-format", "--dump-config", cpp_file],
+        capture_output=True,
+        text=True,
+    )
+    proc.check_returncode()
+    return re.sub(r"(ColumnLimit:\s*)\d+", r"\g<1>{}".format(80), proc.stdout,)
+
+
+def run_clang_format_on_lines(rootdir, changed_file, assume_filename=None):
+    logger = logging.getLogger(__name__)
+
+    line_arguments = [
+        "--lines={}:{}".format(start, end) for start, end in changed_file.lines
+    ]
+    assert line_arguments
+
+    logger.info("Reformatting %s...", changed_file.filename)
+    filename = os.path.join(rootdir, changed_file.filename)
+    cmd = [
+        "clang-format",
+        "--style=file",
+        "--assume-filename={}".format(
+            assume_filename if assume_filename else filename
+        ),
+        *line_arguments,
+    ]
+
+    with open(filename) as fp:
+        logger.debug("Executing: %r", cmd)
+        proc = subprocess.run(cmd, stdin=fp, capture_output=True, text=True)
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError:
+        logger.error(
+            "Error while executing command %s: %s", cmd, proc.stderr,
+        )
+        raise
+
+    if proc.stderr:
+        logger.error(proc.stderr)
+    with open(filename, mode="w+") as fp:
```

Comment:
because w+ truncates the file (in case the reformatted file is shorter than the original file, we would get stray chars at the end)

Reply:
w+ is for reading and writing, but you do not read. 

```
``w''   Truncate file to zero length or create text file for writing.
         The stream is positioned at the beginning of the file.

 ``w+''  Open for reading and writing.  The file is created if it does not
         exist, otherwise it is truncated.  The stream is positioned at
         the beginning of the file.
```